### PR TITLE
Issue: #5953 WIP: Part 1. Add the initial Jenkinsfile for the project

### PR DESCRIPTION
**Pull request by daniilyar**
This Jenkinsfile is required for running Jenkins tests for each PR. It contains the build order and exact build steps definition programmed as a Groovy DSL

### CODE REVIEW

**Comment by daniilyar** [comment]
The Jenkinsfile format was chosen because, thanks to the fact that it's programmatic, more modern and more flexible than old Jenkins manual-UI-configs workflow of configuring builds.
Also keeping this file in repository will ensure that if Jenkins config will be missed, the build workflow definition will not be missed

**Comment by daniilyar** [comment]
@romani I can't fix the commit message because force push is prohibited while I can't create this branch in my fork because I can only test the Jenkins GitHub integration properly via the parent repository.
So later when this work-in-progress will be finished I will re-create the PR with a proper commit message. Thanks!

**Comment by daniilyar** [comment]
Test comment posted via Jenkins

**Comment by romani** [comment]
@daniilyar , please  close this PR and open new PR.
It is better to read PR template.
it is also recommended to run "./.ci/test-spelling-unknown-words.sh"

if you fill that you might not remember all rules ... it is better to do branches in your fork.

**Comment by daniilyar** [comment]
@romani , again, if I will create this in my fork, I will not be able to properly test it from Jenkins. Please read my previous message carefully.

**Comment by daniilyar** [comment]
Ok, I have finished my testing, everything works now (http://jenkins.checkstyle.org/job/pr-builder-checkstyle-pipeline/3/console). Re-creating the PR with a 'good' commit message to satisfy checkers :)

**Comment by daniilyar** [comment]
@romani , well, maybe I didn't state this clearly.. Let me try again. 

Jenkins supports *building PRs coming from forks*, but only when Jenkinsfile is *already present in the target of main repository*. So only at a first step, to test Jenkinsfile before merging it, I needed to put it into a topic branch in the main repo, and not in the fork.

Do you understand now? If you still have questions - let's have a call and I will explain it in more detais

**Comment by romani** [comment]
ok, no problem, lets merge it quicker.
